test/common/network/address_impl_test.cc

```diff
@@ -100,8 +100,10 @@ TEST(Ipv4InstanceTest, BadAddress) {
 
 TEST(Ipv4InstanceTest, SocketBindAndConnect) {
   // Test listening on and connecting to an unused port on the IPv4 loopback address.
-  Ipv4Instance loopbackPort("127.0.0.1", ::Network::Test::getUnusedPort());
-  testSocketBindAndConnect(loopbackPort);
+  InstancePtr addrPort(new Ipv4Instance("127.0.0.1", 0));
+  addrPort = Network::Test::checkPortAvailability(addrPort, SocketType::Stream);
```

Comment:
Suggest renaming - checkFoo implies to me that it might be idempotent or non side-effecting, but this is actually mutating state. Maybe Network::Test::allocatePort?